sds_data_manager/lambda_code/IAlirtCode/ialirt_ingest.py

```diff
@@ -0,0 +1,47 @@
+"""IALiRT ingest lambda."""
+
+import json
+import logging
+import os
+
+import boto3
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def lambda_handler(event, context):
+    """Create metadata and add it to the database.
+
+    This function is an event handler for s3 ingest bucket.
+    It is also used to ingest data to the DynamoDB table.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    context : LambdaContext
+        This object provides methods and properties that provide
+        information about the invocation, function,
+        and runtime environment.
+
+    """
+    logger.info("Received event: %s", json.dumps(event))
+
+    table_name = os.environ.get("TABLE_NAME")
+    dynamodb = boto3.resource("dynamodb")
+    table = dynamodb.Table(table_name)
+
+    s3_filepath = event["detail"]["object"]["key"]
+    filename = os.path.basename(s3_filepath)
+    logger.info("Retrieved filename: %s", filename)
+
+    # TODO: item is temporary and will be replaced with actual packet data.
+    item = {
+        "reset_number#met": "0#123",
```

Comment:
I don't quite understand how you'll get the reset number when writing to the DB since it isn't in the CCSDS packet, or am I mistaken here. I think you can only use MET. Maybe ops-software gets more information out of the "Frame" that wraps the packets which we won't have?

Reply:
To maybe illustrate why we wouldn't want to include the reset number is that you won't want to do two queries in the future for data if we have a reset. Because you wouldn't know what `0#` or `1#` or `2#`, ... to prepend to the query. DynamoDB is more focused on a good key for the retrieval than the initial storage.

Reply:
The reason for the reset_number is that we need to have a unique partition key. In the event that the spacecraft clock is reset then we could have non-unique mets, which would be a big problem. So my initial plan was to change the value of the reset number and redeploy if that happened, but that wouldn't be an automated approach so I think you are right. This need more consideration.

It might be better to put in a sort key of mission date. If the spacecraft had a reset then the combination of partition key and sort key would still be unique. 

I'll take out reset# and put in the mission date as the sort key.